Nodes/xBeeSyncNode.cpp

```diff
@@ -0,0 +1,186 @@
+/****************************************************************************************
+ *
+ * File:
+ * 		xBeeSyncNode.cpp
+ *
+ * Purpose:
+ *		Handles sending of logs or the current system state (xml-format) over xbee,
+ * 		depending on the setting of send_logs. Also receives rudder and sail control messages
+ *		over the xbee network.
+ *
+ * Developer Notes:
+ *
+ *
+ ***************************************************************************************/
+
+#include "xBeeSyncNode.h"
+#include <sstream>
+
+xBeeSyncNode::xBeeSyncNode(MessageBus& msgBus, DBHandler* db) :
+	ActiveNode(NodeID::xBeeSync, msgBus), m_db(db)
+{
+
+}
+bool xBeeSyncNode::init()
+{
+
+	m_sending = m_db->retrieveCellAsInt("xbee_config", "1", "send");
+	m_receiving = m_db->retrieveCellAsInt("xbee_config", "1", "recieve");
+	m_sendLogs = m_db->retrieveCellAsInt("xbee_config", "1", "send_logs");
+	m_loopTime = stod(m_db->retrieveCell("xbee_config", "1", "loop_time"));
+	m_pushOnlyLatestLogs = m_db->retrieveCellAsInt("xbee_config", "1", "push_only_latest_logs");
+
+	bool rv = false;
+
+	m_xbee_fd = m_xBee.init();
+
+	if(m_xbee_fd < 0)
+	{
+		Logger::error("XbeeSync::%d Failed to initalise", __LINE__);
+	}
+	else
+	{
+		rv = true;
+		Logger::info("Xbee initialised - receiving: %d sending: %d", m_receiving, m_sending);
+		m_initialised = true;
+	}
+
+	return rv;
+}
+
+void xBeeSyncNode::start(){
+
+	if (m_initialised)
+    {
+
+		m_messageTimeBuffer = m_loopTime;
+        runThread(xBeeSyncThread);
+    }
+    else
+    {
+        Logger::error("%s Cannot start XBEESYNC thread as the node was not correctly initialised!", __PRETTY_FUNCTION__);
+    }
+
+}
+
+void xBeeSyncNode::processMessage(const Message* msgPtr)
+{
+    MessageType msgType = msgPtr->messageType();
+
+    switch(msgType)
+    {
+		case MessageType::VesselState:
+			sendVesselState((VesselStateMsg*)msgPtr);
+			break;
+        default:
+            break;
+    }
+
+}
+
+void xBeeSyncNode::sendVesselState(VesselStateMsg* msg){
+
+	//Do not allow sending of vesselstate and logs at the same time; double output turns into mush
+	if (!m_sendLogs && m_sending)
+	{
+		//The amount of seconds between expected calls from VesselStateMsg
+		double stateMessageInterval = 0.4;
```

Comment:
Use the timer class for more accurate measurement or SysClock::UnixTime, this will also mean that if the VesselStateNode pumps messages out at different times in the future you won't have to change this class.